File: Signiture-Backend/delivery/db/src/main/java/pcy/study/db/userordermenu/UserOrderMenuRepository.java
package pcy.study.db.userordermenu;

import org.springframework.data.jpa.repository.JpaRepository;
import pcy.study.db.userordermenu.enums.UserOrderMenuStatus;

import java.util.List;

public interface UserOrderMenuRepository extends JpaRepository<UserOrderMenu, Long> {

    List<UserOrderMenu> findAllByUserOrderIdAndStatusOrderByIdDesc(Long userOrderId, UserOrderMenuStatus status);
}

File: Signiture-Backend/kotlin/delivery/db/src/main/kotlin/pcy/study/db/userordermenu/UserOrderMenuRepository.kt
package pcy.study.db.userordermenu

import org.springframework.data.jpa.repository.JpaRepository
import pcy.study.db.userorder.UserOrder
import pcy.study.db.userordermenu.enums.UserOrderMenuStatus

interface UserOrderMenuRepository : JpaRepository<UserOrderMenu, Long> {

    fun findAllByUserOrderAndStatusOrderByIdDesc(
        userOrder: UserOrder?,
        status: UserOrderMenuStatus?
    ): List<UserOrderMenu>
}
